pkgs/development/compilers/bluespec/default.nix

```diff
@@ -0,0 +1,94 @@
+{ stdenv
+, fetchFromGitHub
+, fetchpatch
+, autoconf
+, automake
+, fontconfig
+, gmp
+, gperf
+, haskell
+, libX11
+, libpoly
+, perl
+, pkgconfig
+, verilog
+, xorg
+, zlib
+}:
+
+let
+  # yices wants a libgmp.a and fails otherwise
+  gmpStatic = gmp.override { withStatic = true; };
+
+  # Compiling PreludeBSV fails with more recent GHC versions
+  # > imperative statement (not BVI context)
+  # https://github.com/B-Lang-org/bsc/issues/20#issuecomment-583724030
+  ghcWithPackages = haskell.packages.ghc844.ghc.withPackages (g: (with g; [old-time regex-compat syb]));
+in stdenv.mkDerivation rec {
+  pname = "bluespec";
+  version = "unstable-2020.02.09";
```

Comment:
I don't think this is a good version string, because even if `lib.compareVersion` works against it, many tools would generally expect version identifiers to begin numerically. I do, at least.

Here's my suggestion: rather than inventing this, let's just bite the bullet and follow something close to calver/semver for this. Historically, Bluespec compiler releases were tagged as `YEAR.MONTH` e.g. `2014.08`, which looks fine. So then let's extend this to support the metadata/prerelease notions of that:

```
version = "2020.02-unstable+g{builtins.substr 0 8 src.rev}
```

Where `-unstable` signifies the prerelease identifier (it might one day be `-beta2` or `-rc3` for example), and `+g$HASH` signifies the build metadata containing the git version, like in semver. But the basic major version numer is still calendar versioning.

In fact I normally even use a slightly augmented version of this, inspired by Nix itself, which tracks the number of commits to the head that you target. So it would actually, in this case, be:

```
vcsRev = "68"; # 68 commits as of 05c8afb08078e437
version = "2020.02-unstable+${vcsRev}-g{builtins.substr 0 8 src.rev}
```

Which would look something like `bluespec-2020.02-unstable+68+05c8afb0` which is wordy, but very accurate. (The `68` can be generated via `git log --format=oneline | wc -l`)

In the event of a stable release, we'd just change it to e.g. `version = "2020.03"` or something.

Does this sound reasonable?

Reply:
I'm just following https://github.com/NixOS/nixpkgs/blob/master/doc/contributing/coding-conventions.xml#L232 here, and don't see why we should do any different here.

Apart from that, I personally really like just using `git describe --tags` for version strings - but this would require upstream tag at least a single commit with something like `2020.02`.

Reply:
I proposed this to upstream already in https://github.com/B-Lang-org/bsc/issues/13.